src/hooks/core/chat/useFetchPostNotificationHook.ts

```diff
@@ -13,24 +14,33 @@ const useFetchPostNotificationHook = () => {
   const saveNotifications = async (notifications: any[], fromNotificationAPI: any) => {
     if (!localDb) return;
 
-    notifications.map((notification) => {
-      queue.addJob({
-        label: 'save post notifications',
-        task: async () => {
-          const channelList = fromNotificationAPI(notification);
-          await channelList.save(localDb);
-        }
-      });
+    notifications.map(async (notification) => {
+      if (Platform.OS === 'android') {
+        const channelList = fromNotificationAPI(notification);
+        await channelList.save(localDb);
+      } else {
+        queue.addJob({
+          label: 'save post notifications',
+          task: async () => {
+            const channelList = fromNotificationAPI(notification);
+            await channelList.save(localDb);
+          }
+        });
+      }
 
       return null;
     });
 
-    queue.addJob({
-      label: 'refresh channelList',
-      task: async () => {
-        refresh('channelList');
-      }
-    });
+    if (Platform.OS === 'android') {
+      refresh('channelList');
+    } else {
+      queue.addJob({
+        label: 'refresh channelList',
+        task: async () => {
+          refresh('channelList');
+        }
+      });
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here introduce platform-specific logic for saving notifications and refreshing the channel list. This is done by checking if the platform is Android and then either directly executing the tasks or adding them to a job queue. However, there's an issue with the use of `Array.prototype.map()` method. The `map()` method is used to create a new array with the results of calling a provided function on every element in the array. Here, you're not using the result of the `map()` method, so it would be more appropriate to use `Array.prototype.forEach()` instead. Also, since you're performing asynchronous operations inside the loop, you should consider using `Promise.all()` to ensure all promises are resolved before proceeding.

```diff
-    notifications.map(async (notification) => {
+    await Promise.all(notifications.map(async (notification) => {
      if (Platform.OS === 'android') {
        const channelList = fromNotificationAPI(notification);
        await channelList.save(localDb);
      } else {
        queue.addJob({
          label: 'save post notifications',
          task: async () => {
            const channelList = fromNotificationAPI(notification);
            await channelList.save(localDb);
          }
        });
      }
  
-       return null;
-     });
+    }));
  
    if (Platform.OS === 'android') {
      refresh('channelList');
    } else {
      queue.addJob({
        label: 'refresh channelList',
        task: async () => {
          refresh('channelList');
        }
      });
    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->